service/controller/resource/app/desired.go

```diff
@@ -58,7 +59,32 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*g8s
 		userConfig := newUserConfig(cr, appSpec, configMaps, secrets)
 
 		if !appSpec.LegacyOnly {
-			apps = append(apps, r.newApp(*cc, cr, appSpec, userConfig))
+			app := r.newApp(*cc, cr, appSpec, userConfig)
+
+			// For chart-operator only we need to set the Helm major version
+			// label. This is so the correct instance of app-operator processes
+			// the CR.
+			if appSpec.App == chartOperatorAppName {
+				var helmMajorVersion string
+
+				{
+					version, err := semver.NewVersion(appSpec.Version)
+					if err != nil {
+						return nil, microerror.Mask(err)
+					}
+
+					// For chart-operator 1.0.0 and greater we use Helm 3.
+					if version.Major() >= 1 {
+						helmMajorVersion = "3"
+					} else {
+						helmMajorVersion = "2"
+					}
```

Comment:
I know sometimes we like to have fewer indentation blocks. 
```suggestion
                                            helmMajorVersion = "2"
					if version.Major() >= 1 {
						helmMajorVersion = "3"
					}
```

Reply:
Yes sure. I tried to simplify it and removed the block as this is already in a conditional. 

PTAL